### PR TITLE
Update to 0.638 and cleanup build script

### PR DESCRIPTION
**Pull request by RyeMutt**
* Update to Luau 0.638 and switch to submodule vendoring
* Clean up build script to generate windows symbols for static libraries
* Fix macOS targeting
* Add test runs

### CODE REVIEW

**Review by bennettgoble — COMMENTED** [review]
LGTM! Thanks.

https://github.com/secondlife/3p-luau/releases/tag/v0.638-r2

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -1,22 +1,30 @@
...
+        if: ${{ runner.os == 'Linux' }}
+        shell: bash
+        run: |
+          sudo apt-get update
+          sudo apt-get -y install ninja-build
+      - uses: secondlife/action-autobuild@v4
         with:
-          addrsize: "64"
-
+          addrsize: ${{ matrix.addrsize }}
```
Comment (nat-goodspeed) [inline .github/workflows/build.yaml:27]: Are we contemplating building packages for multiple address sizes again?

File: build-cmd.sh [hunk 2]
```diff
@@ -50,31 +47,102 @@ pushd "$top/luau"
...
         windows*)
             set -o igncr
-            opts="$LL_BUILD_RELEASE /EHsc"
-            cmake -G "$AUTOBUILD_WIN_CMAKE_GEN" -A "$AUTOBUILD_WIN_VSPLATFORM" \
+            opts="$(replace_switch /Zi /Z7 $LL_BUILD_RELEASE) /EHsc"
+            plainopts="$(remove_switch /GR $(remove_cxxstd $opts))"
+
+            cmake -G "Ninja" -DCMAKE_BUILD_TYPE="Release" \
                   -DCMAKE_INSTALL_PREFIX="$(cygpath -m "$stage")" \
                   -DCMAKE_C_FLAGS="$(remove_cxxstd $opts)" \
```
Comment (nat-goodspeed) [inline build-cmd.sh:55]: Did you intend to change to `-DCMAKE_C_FLAGS="$plainopts"` ? I'm not sure why else to introduce `plainopts` at line 51; there's no other reference to it in the `windows*` stanza.

File: build-cmd.sh [hunk 3]
```diff
@@ -50,31 +47,102 @@ pushd "$top/luau"
...
+                ./Luau.Conformance.exe -O2
+                ./Luau.Conformance.exe -O2 --fflags=true
+                ./Luau.Conformance.exe --codegen
+                ./Luau.Conformance.exe --codegen --fflags=true
+                ./Luau.Conformance.exe --codegen -O2
+                ./Luau.Conformance.exe --codegen -O2 --fflags=true
+                ./luau ../luau/tests/conformance/assert.lua
+                ./luau-analyze ../luau/tests/conformance/assert.lua
+                ./luau-compile ../luau/tests/conformance/assert.lua
+            fi
```
Comment (nat-goodspeed) [inline build-cmd.sh:75]: NICE. Thanks.

File: build-cmd.sh [hunk 4]
```diff
@@ -50,31 +47,102 @@ pushd "$top/luau"
...
-            cp -v "Release/Luau.VM.lib" "$stage/lib/release/"
+            cp -v "Luau.Ast.lib" "$stage/lib/release/"
+            cp -v "Luau.CodeGen.lib" "$stage/lib/release/"
+            cp -v "Luau.Compiler.lib" "$stage/lib/release/"
+            cp -v "Luau.Config.lib" "$stage/lib/release/"
+            cp -v "Luau.VM.lib" "$stage/lib/release/"
+
+            cp -v luau.exe "$stage/bin/"
+        ;;
+        darwin*)
```
Comment (nat-goodspeed) [inline build-cmd.sh:87]: So much of this is replicated between the three platform stanzas, it seems as though after the specific build commands, the rest could be moved to the common code following `esac`.

Of course you'd need platform-specific
```
exe='.exe'
function lib {
    echo "$1.lib"
}
```
vs.
```
exe=''
function lib {
    echo "lib$1.a"
}
```
to unify the differences in the replicated commands.

While we're at it, though, why build `luau.exe` only on Windows? I've found it useful to be able to run the standalone REPL on Mac too.

**Pull request merged**